src/map/ai/states/item_state.cpp

```diff
@@ -0,0 +1,303 @@
+﻿/*
+===========================================================================
+
+Copyright (c) 2010-2015 Darkstar Dev Teams
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see http://www.gnu.org/licenses/
+
+This file is part of DarkStar-server source code.
+
+===========================================================================
+*/
+
+#include "item_state.h"
+
+#include "../ai_container.h"
+#include "../../entities/battleentity.h"
+#include "../../entities/charentity.h"
+
+#include "../../item_container.h"
+#include "../../recast_container.h"
+#include "../../status_effect_container.h"
+#include "../../universal_container.h"
+
+#include "../../packets/action.h"
+#include "../../packets/inventory_assign.h"
+#include "../../packets/inventory_item.h"
+#include "../../packets/inventory_finish.h"
+
+#include "../../utils/battleutils.h"
+#include "../../utils/charutils.h"
+#include "../../../common/utils.h"
+
+
+CItemState::CItemState(CBattleEntity* PEntity, uint16 targid, uint8 loc, uint8 slotid) :
+    CState(PEntity, targid),
+    m_PEntity(PEntity),
+    m_PItem(nullptr)
+{
+    auto PChar = static_cast<CCharEntity*>(m_PEntity);
+    auto PItem = dynamic_cast<CItemUsable*>(PChar->getStorage(loc)->GetItem(slotid));
+
+    if (PItem && PItem->isType(ITEM_USABLE))
+    {
+        if (PItem->isType(ITEM_ARMOR))
+        {
+            // todo: if item is locked, check if its equipped
+        }
+        else if (PItem->isSubType(ITEM_LOCKED))
+        {
+            return;
```

Comment:
throw exception instead of returning if there's an error (or else the state will construct and run like normal)